packages/main/src/plugin/navigation/search-result-provider-registry.ts

```
@@ -0,0 +1,104 @@
+      });
+    }
+
+    this.#providers.set(id, { id, label, extensionIcon, provider });
+
+    return Disposable.create(() => {
+      this.#providers.delete(id);
+      changeDisposable?.dispose();
+    });
+  }
+
+  async search(query: string, maxResults: number = 10): Promise<SearchResultItemInfo[]> {
```

Biome (2.5.6)

[error] This type annotation is trivially inferred from its initialization.

(lint/style/noInferrableTypes)